libraries/core_libs/consensus/src/transaction/transaction_manager.cpp

```diff
@@ -297,21 +303,22 @@ std::vector<std::shared_ptr<Transaction>> TransactionManager::getNonfinalizedTrx
  */
 std::pair<SharedTransactions, std::vector<uint64_t>> TransactionManager::packTrxs(uint64_t proposal_period,
                                                                                   uint64_t weight_limit) {
-  std::shared_lock transactions_lock(transactions_mutex_);
   SharedTransactions trxs;
   std::vector<uint64_t> estimations;
-  transactions_pool_.resetGetNextTransactionIterator();
   uint64_t total_weight = 0;
-  while (true) {
-    auto trx = transactions_pool_.getNextTransaction();
-    if (trx == nullptr) break;
-    uint64_t weight = estimateTransactionGas(trx, proposal_period);
+  {
+    std::shared_lock transactions_lock(transactions_mutex_);
+    uint64_t max_transactions_in_block = weight_limit / kMinTxGas;
```

Comment:
this could be const and maybe not under lock?

Reply:
Set to const and moved out